openmc/search.py

```diff
@@ -193,12 +197,26 @@ def search_for_keff(model_builder, initial_guess=None, target=1.0,
 
     # Add information to be passed to the searching function
     args['args'] = (target, model_builder, model_args, print_iterations,
-                    run_args, guesses, results)
+                    run_args, guesses, results, run_in_memory)
 
     # Create a new dictionary with the arguments from args and kwargs
     args.update(kwargs)
 
     # Perform the search
-    zero_value = root_finder(**args)
+    if not run_in_memory:
+        zero_value = root_finder(**args)
+        return zero_value, guesses, results
 
-    return zero_value, guesses, results
+    else:
+        try:
+            zero_value, root_res = root_finder(**args, full_output=True, disp=False)
+            if root_res.converged:
+                return zero_value, guesses, results
+
+            else:
+                print(f'WARNING: {root_res.flag}')
+                return guesses, results
+        # In case the root finder is not successful
+        except Exception as e:
+            print(f'WARNING: {e}')
+            return guesses, results
```

Comment:
@church89 I'm interested in getting feedback from other people on these changes. My personal preference is having a consistent return signature (always three values) rather than sometimes two, sometime three.

But, moving to returning three values is a large API change that would benefit from discussion with other users and developers.

The decision needs to also be reflected in the documentation for this function: when three values are returned or not, or adding the `zero_value` to the return block

ping @paulromano @pshriwise @shimwell or whomever else